modules/renter/hostdb/scan.go

```diff
@@ -254,6 +194,26 @@ func (hdb *HostDB) threadedProbeHosts() {
 		case <-hdb.tg.StopChan():
 			return
 		case hostEntry := <-hdb.scanPool:
+			// Block the scan until the host is online.
```

Comment:
I may not have pushed the code yet, but they get inserted into the host tree immediately upon being discovered in the blockchain.

A full rescan won't occur at this point in the code though. By the time any of the scan loops have launched, any hdb startup like a rescan has already completed.